doc/features/tuning-policies/README.md

```diff
@@ -143,20 +143,24 @@ A retry policy centralizes the handling of query retries, minimizing the need fo
 
 The retry policy interface consists of four methods:
 
-- `#onReadTimeout(requestInfo, consistency, received, blockFor, isDataPresent)`: determines what to do when the driver
+- `#onReadTimeout(info, consistency, received, blockFor, isDataPresent)`: determines what to do when the driver
 gets a `ReadTimeoutException` response from a Cassandra node.
-- `#onUnavailable(requestInfo, consistency, required, alive)`: determines what to do when the driver gets an 
+- `#onUnavailable(info, consistency, required, alive)`: determines what to do when the driver gets an 
 `UnavailableException` response from a Cassandra node.
-- `#onWriteTimeout(requestInfo, consistency, received, blockFor, writeType)`: determines what to do when the driver gets
+- `#onWriteTimeout(info, consistency, received, blockFor, writeType)`: determines what to do when the driver gets
 a `WriteTimeoutException` response from a Cassandra node
-- `#onRequestError(requestInfo, consistency, err)`: defines whether to retry and at which consistency level on an 
+- `#onRequestError(info, consistency, err)`: defines whether to retry and at which consistency level on an 
 unexpected error, invoked in the following situations:
     - On a client timeout, while waiting for the server response , being the error an instance of 
     `OperationTimedOutError`.
     - On a connection error (socket closed, etc.).
     - When the contacted host replies with an error, such as `overloaded`, `isBootstrapping`, `serverError`, etc. In 
     this case, the error is instance of `ResponseError`
 
-A default and base retry policy is included.
+A default and base retry policy is included, along with `IdempotenceAwareRetryPolicy` that considers.
```

Comment:
I think there is a risk of breaking user code, as we mark all queries non-idempotent by default the driver won't sent use the retry policy for all cases.
I think going through the same transition (expose `IdempotenceAwareRetryPolicy` initially) as the java driver did is the safer route. We can change to the desired behaviour (ignore retrying logic for non-idempotent queries) on a major version. 

Being conservative with behavioural changes now will allow us to only maintain the latest 3.x branch, providing patches only to the latest minor and advising users to upgrade to the latest version (within the same major). Once we accumulate enough breaking changes (and the timing is right), we can provide a new major and maintain both 3.x and 4.x branches.

I've created a ticket for the breaking change: https://datastax-oss.atlassian.net/browse/NODEJS-331.